project_search.py

```diff
@@ -0,0 +1,147 @@
+    for runfolder in runfolders:
+        projpath = os.path.join(runfolder, "Unaligned", project)
+        for i in [1, 2]:
+            fq = glob(os.path.join(projpath, "**", f"*R{i}*fastq.gz"), recursive=True)
+            fastqs[runfolder][f"R{i}"] = [os.path.basename(path) for path in fq]
+            fastqs["Total"][f"R{i}"] += len(fq)
+    return fastqs
+
+def check_organization(fastqs, org_paths):
+    not_organized = {}
+    for runfolder_path, fq_filenames in fastqs.items():
+        if runfolder_path == "Total":
+            continue
+        runfolder = os.path.basename(runfolder_path)
+        for org_path in org_paths:
+            org_dir = os.path.basename(org_path)
+            for i in [1, 2]:
+                org_fq_paths = glob(os.path.join(org_path, "**", f"*R{i}*fastq.gz"), recursive=True)
+                org_fq = {(os.path.basename(os.path.dirname(path)), os.path.basename(path)) for path in org_fq_paths}
+                for fq_filename in fq_filenames[f"R{i}"]:
+                    if (runfolder, fq_filename) not in org_fq:
+                        not_organized.setdefault(org_dir, [])
+                        not_organized[org_dir].append((runfolder, fq_filename))
+    return not_organized
+
+def main():
+    parser = argparse.ArgumentParser(description="Find runfolders containing a specific project ID.")
+    parser.add_argument("--project", required=True, help="The project ID to search for.")
+    parser.add_argument("--check_org", required=False, action="store_true", help="Check if organization(s) in /proj/ngi2016001/nobackup/DATA/<project>* corresponds to identified samples and runfolders")
+    parser.add_argument("--list_files", required=False, help="List all fastq files that are not organized in specified folder. Requires --check_org")
+    args = parser.parse_args()
+    project = args.project
+    check_org = args.check_org
+    list_files = args.list_files
+    runfolders = find_runfolders_with_project(project)
+
+    if len(runfolders) == 0:
+        print(f"{project} could not be identified in any runfolder.")
+        return
+
+    sample_info = parse_samplesheet(runfolders, project)
+    
+    fastqs = get_fastqs(runfolders, project)
+    
+    print_result(runfolders, sample_info, fastqs)
+
+    if check_org:
+        org_paths = os.path.join(r"/proj/ngi2016001/nobackup/NGI/DATA/", f"{project}*")
+        org_paths = glob(org_paths)
+        print("\nChecking organization in\n" + "\n".join(org_paths))
+        not_org = check_organization(fastqs, org_paths)
+        if len(not_org) == 0:
+            for org_path in org_paths:
+                org_dir = os.path.basename(org_path)
+                print(f"\nAll fastq files are organized for analysis in {org_dir}")
+        else:
+            print("")
+            for org_dir in not_org:
+               print(f"{len(not_org[org_dir])} fastq files (R1 + R2) are not organized in {org_dir}")
+            if list_files:
```

Comment:
Isn't it always useful to list the files? I wonder if this parameter really is needed?